twine/exceptions.py

```diff
@@ -13,33 +14,52 @@
 # limitations under the License.
 
 
-class RedirectDetected(Exception):
+class TwineException(Exception):
```

Comment:
As we're working towards an API. I've updated all of Twine's exceptions to have a single root in `TwineException`. This makes API users' lives easier. They can use a catch-all

```py
try:
    ...
except twine.exceptions.TwineException:
    ...
```

Instead of trying to catch all of them. This can be removed from this PR if desired.